Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/RuntimeCompletionExpressionType.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+ * for details. All rights reserved. Use of this source code is governed by a
+ * BSD-style license that can be found in the LICENSE file.
+ *
+ * This file has been automatically generated.  Please do not edit it manually.
+ * To regenerate the file, use the script "pkg/analysis_server/tool/spec/generate_files".
+ */
+package org.dartlang.analysis.server.protocol;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import com.google.common.collect.Lists;
+import com.google.dart.server.utilities.general.JsonUtilities;
+import com.google.dart.server.utilities.general.ObjectUtilities;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import java.util.ArrayList;
+import java.util.Iterator;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * A type at runtime.
+ *
+ * @coverage dart.server.generated.types
+ */
+@SuppressWarnings("unused")
+public class RuntimeCompletionExpressionType {
+
+  public static final RuntimeCompletionExpressionType[] EMPTY_ARRAY = new RuntimeCompletionExpressionType[0];
+
+  public static final List<RuntimeCompletionExpressionType> EMPTY_LIST = Lists.newArrayList();
+
+  /**
+   * The path of the library that has this type. Omitted if the type is not declared in any library,
+   * e.g. "dynamic", or "void".
+   */
+  private final String libraryPath;
```

Comment:
For easy integration with the observatory it would be convenient if this functioned ok if a path to the actual source file defining the class was ok rather than strictly the library path. it appears the analyzer does fine with imports to part files so I would expect that isn't much of an issue to support and it is just a matter of documenting that this can be a path to the library or the actual part file in the library defining the type.

Reply:
Does Observatory provide only the file that defines a class, but not the library?

We can change it to the file path, analyzer knows which libraries include which parts.